src/Cafe/IOSU/legacy/iosu_fpd.cpp

```diff
@@ -1143,6 +1147,31 @@ namespace iosu
 				return FPResult_Ok;
 			}
 
+			nnResult CallHandler_UpdateCommentAsync(FPDClient* fpdClient, IPCIoctlVector* vecIn, uint32 numVecIn, IPCIoctlVector* vecOut, uint32 numVecOut)
+			{
+				std::unique_lock _l(g_fpd.mtxFriendSession);
+				if (numVecIn != 1 || numVecOut != 0)
+					return FPResult_InvalidIPCParam;
+				if (!g_fpd.nexFriendSession)
+					return FPResult_RequestFailed;
+				DeclareInputPtr(newComment, char16_t, (vecOut[0].size / 2), 0);
+				IPCCommandBody* cmd = ServiceCallDelayCurrentResponse();
+
+				auto utf8_comment = StringHelpers::ToUtf8((const uint16be*)newComment, vecIn[0].size);
```

Comment:
You are passing the size in bytes to `ToUtf8`, but it wants the size in wide characters.